service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
+}
+
+func allInstancesPresent(instances []*ec2.Instance) bool {
+	for _, instance := range instances {
+		if instance == nil {
+			return false
+		}
+	}
+
+	return true
```

Comment:
I would like to see tests for this. When the list is empty it returns true and the code above ala `masters[0]` panics. This is what we want to prevent.